src/food/food.service.ts

```diff
@@ -117,107 +132,100 @@ export class FoodService {
 
-    // hotLevel은 있지만 category가 없는 경우
-    if (!category) {
+        return sortBy(query, sort)
+          .take(size)
+          .getMany()
+          .then(produceFindFoodDto);
+      }
       const query = this.foodRepository
         .createQueryBuilder('food')
+        .leftJoinAndSelect('food.categories', 'category')
         .leftJoinAndSelect('food.foodLevel', 'foodLevel')
-        .where('foodLevel.id = :hotLevelId', { hotLevelId });
+        .where('category.name = :categoryName', { categoryName: category })
+        .andWhere('foodLevel.id = :hotLevelId', { hotLevelId });
 
       return sortBy(query, sort).take(size).getMany().then(produceFindFoodDto);
+    } catch (e) {
+      throw new HttpException(ERROR_MESSAGE.NOT_FOUND, HttpStatus.BAD_REQUEST);
     }
-    const query = this.foodRepository
-      .createQueryBuilder('food')
-      .leftJoinAndSelect('food.categories', 'category')
-      .leftJoinAndSelect('food.foodLevel', 'foodLevel')
-      .where('category.name = :categoryName', { categoryName: category })
-      .andWhere('foodLevel.id = :hotLevelId', { hotLevelId });
-
-    return sortBy(query, sort).take(size).getMany().then(produceFindFoodDto);
   }
 
   // 레벨 별 음식 정보를 가져 옵니다.
-  async findUserLevelFoods(param): Promise<Food[]> {
-    const { userLevel } = param;
-
-    if (Number(userLevel) < 1 || Number(userLevel) > 4) {
-      return [];
-    }
-
-    const { id: foodLevel } = await this.foodLevelRepository
+  async findUserLevelFoods(userLevel): Promise<FoodLevel> {
+    const foodLevel = await this.foodLevelRepository
       .createQueryBuilder('foodLevel')
       .leftJoinAndSelect('foodLevel.userLevel', 'userLevel')
       .where('foodLevel.userLevel = :userLevel', { userLevel })
       .getOne();
 
-    return await this.foodRepository
+    return foodLevel;
+  }
+
+  async findTreeFood(foodLevel): Promise<Food[]> {
+    const foods: Food[] = await this.foodRepository
       .createQueryBuilder('food')
       .leftJoinAndSelect('food.foodLevel', 'foodLevel')
-      .select([
-        'food.id',
-        'food.name',
-        'food.subName',
-        'food.imageUrl',
-        'food.logoImageUrl',
-      ])
+      .select(['food.id', 'food.name', 'food.subName', 'food.imageUrl'])
       .where('food.foodLevel = :foodLevel', { foodLevel })
       .orderBy('RAND()')
       .limit(3)
       .getMany();
+    return foods;
   }
 
-  async findRandomFoods(userId): Promise<RandomFoodDto> {
+  async findUserLevel(userId): Promise<UserLevel> {
     const { userLevel: userlevel } = await this.userRepository
       .createQueryBuilder('user')
       .leftJoinAndSelect('user.userLevel', 'userLevel')
       .select()
       .where('user.id = :userId', { userId })
       .getOne();
+    return userlevel;
+  }
 
-    if (userlevel.id === '5') {
-      userlevel.id = '4';
-    }
-
-    return await this.foodRepository
+  async findFoodByUserLevel(userlevel): Promise<Food> {
+    const food = await this.foodRepository
```

Comment:
```suggestion
  async findFoodByUserLevel(userlevel): Promise<Food> {
    return this.foodRepository
```

이렇게 변수에 할당하지 않고 return하는 것이 메모리 낭비가 적습니다~